### PR TITLE
Loop ranges should handle from = to, and generate one iteration.

### LINKED ISSUE #935: Loop.range generates inconsistent number of iterations when start=end
/kind bug

**What steps did you take and what happened:**

Reproduce using the following code

``` python
from kfp import dsl, components
from kfp_tekton.tekton import Loop

op1_yaml = '''\
name: 'my-in-coop1'
inputs:
- {name: item, type: Integer}
- {name: param, type: Integer}
implementation:
    container:
        image: library/bash:4.4.23
        command: ['sh', '-c']
        args:
        - |
          set -e
          echo op1 "$0" "$1"
        - {inputValue: item}
        - {inputValue: param}
'''


@dsl.pipeline(name='pipeline')
def pipeline():
    with Loop.range(start=0, step=1, end=0) as item:
        op1_template = components.load_component_from_text(op1_yaml)
        op1 = op1_template(item, 10)

if __name__ == '__main__':
  from kfp_tekton.compiler import TektonCompiler as Compiler
  from kfp_tekton.compiler.pipeline_utils import TektonPipelineConf
  tekton_pipeline_conf = TektonPipelineConf()
  tekton_pipeline_conf.set_tekton_inline_spec(True)
  # tekton_pipeline_conf.set_resource_in_separate_yaml(True)
  Compiler().compile(pipeline, __file__.replace('.py', '.yaml'), tekton_pipeline_conf=tekton_pipeline_conf)
```

This generates 0 iterations.

**What did you expect to happen:**
I expect this to generate 1 iteration given that the loop `Loop.range(start=0, step=1, end=1)` generates 2 iterations.

**Additional information:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

* Python Version (use `python --version`):
* SDK Version: 
* Tekton Version (use `tkn version`):
* Kubernetes Version (use `kubectl version`):
* OS (e.g. from `/etc/os-release`):

### PR DESCRIPTION
**Pull request by ScrapCodes**
**Which issue is resolved by this Pull Request:** 
Resolves #935 

**Description of your changes:**
Where start = 0 and end = 0, generate one iteration.
**Environment tested:**

* Python Version (use `python --version`):
* Tekton Version (use `tkn version`):
* Kubernetes Version (use `kubectl version`):
* OS (e.g. from `/etc/os-release`):

**Checklist:**
- [ ] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

File: tekton-catalog/pipeline-loops/pkg/reconciler/pipelinelooprun/pipelinelooprun.go [hunk 1]
```diff
@@ -892,6 +892,11 @@ func computeIterations(run *v1alpha1.Run, tls *pipelineloopv1alpha1.PipelineLoop
 			}
 		}
 	}
+	if from == to && to == 0 && step != 0 {
```
Comment (Tomcli) [inline tekton-catalog/pipeline-loops/pkg/reconciler/pipelinelooprun/pipelinelooprun.go:895]: I think the use case is when from == to, then the iteration is equals to 1. To doesn't have to be 0. 

Another edge case could be that `to` - `from` < `step` if step > 0 and `to` - `from` > `step` for step < 0, then the iteration is also 1.

File: tekton-catalog/pipeline-loops/pkg/reconciler/pipelinelooprun/pipelinelooprun_test.go [hunk 2]
```diff
@@ -1853,7 +1853,7 @@ func TestReconcilePipelineLoopRun(t *testing.T) {
 		pipelineruns:         []*v1beta1.PipelineRun{},
 		expectedStatus:       corev1.ConditionUnknown,
 		expectedReason:       pipelineloopv1alpha1.PipelineLoopRunReasonRunning,
-		expectedPipelineruns: []*v1beta1.PipelineRun{expectedParaPipelineRun, expectedParaPipelineRun1},
+		expectedPipelineruns: []*v1beta1.PipelineRun{expectedParaPipelineRun},
```
[the remark this answers is not in the document]
Comment (Tomcli) [reply tekton-catalog/pipeline-loops/pkg/reconciler/pipelinelooprun/pipelinelooprun_test.go:1856]: Are we no longer checking the `expectedParaPipelineRun1` pipeline? If so we should also remove the `expectedParaPipelineRun1` variable.

[the remark this answers is not in the document]
Comment (ScrapCodes) [reply tekton-catalog/pipeline-loops/pkg/reconciler/pipelinelooprun/pipelinelooprun_test.go]: This was fixed, by not picking default values for, from and to parameters.

**Comment by Tomcli** [comment]
thanks @ScrapCodes 
/lgtm

**Pull request merged**